lib/modules/dosomething/dosomething_reportback/includes/dosomething_reportback_file.inc

```diff
@@ -58,17 +58,16 @@ class ReportbackFileEntity extends Entity {
       // Store that instead.
       $this->review_source = $values['source'];
     }
-    if ($this->status === 'flagged') {
```

Comment:
Maybe renaming the function to `setFlagged` instead of `flag`.. and moving that status conditional there? If the status is `flagged`, then it should be 1 else set it to 0 ?  Not sure without testing but I think that would still allow reviewers the ability to correct any mistakes..